Assets/Scripts/Army/Army.cs

```diff
@@ -6,41 +6,107 @@
 using UnityEngine;
 using UnityEngine.Events;
 using Mirror;
+using Unity.VisualScripting;
 
 [RequireComponent(typeof(ArmyMovement))]
 public class Army : Entity
 {
     readonly SyncList<Unit> armyUnits = new SyncList<Unit>(); // Change to set if necessary
+    
+    // the mean of the army identity on complex plane
+    // this is calculated on on unit add/remove
+    private Vector2 _meanZ;
+    // the complex version of the armies
+    // TODO: please refactor this to be a part of identity info
+    // this is updated per frame
+    private List<Vector2> _armyComplex;
+
+    // this might need to be changed
+    private float _identityChangeRate = 0.2f;
+
     public ReadOnlyCollection<Unit> ArmyUnits => new ReadOnlyCollection<Unit>(armyUnits);
     
     public Army() { }
 
     public void AddUnit(Unit unit)
     {
+        var z = CalculateIdentityComplex(unit);
+        if (armyUnits.Count == 0)
+        {
+            _meanZ = z;
+        }
+        else
+        {
+            _meanZ = _meanZ * (float) armyUnits.Count + z;
+            _meanZ /= (armyUnits.Count + 1);
+        }
+        
         armyUnits.Add(unit);
     }
+    
     public void RemoveUnit(Unit unit)
     {
+        var z = CalculateIdentityComplex(unit);
+        if (armyUnits.Count == 0)
+        {
+            _meanZ = Vector2.zero;
+        }
+        else
+        {
+            _meanZ = _meanZ * (float) armyUnits.Count - z;
+            _meanZ /= (armyUnits.Count - 1);
+        }
         armyUnits.Remove(unit);
     }
+    
     public void SetUnits(Unit[] units) // Use array because Mirror doesn't support lists in commands 
     {
         armyUnits.Clear();
+        _meanZ = Vector2.zero;
         foreach(Unit unit in units)
         {
-            armyUnits.Add(unit);
+            AddUnit(unit);
         }
     }
+
+    private void FixedUpdate()
+    {
+        // I really don't want to calculate this multiple times per frame
+        // ideally this is refactored into IdentityInfo
+        _armyComplex = IdentityComplex();
+        
+        // this is the mean H that all others will tend towards
+        var meanAngle = Mathf.Atan2(_meanZ.y, _meanZ.x);
+        for (int i = 0; i < armyUnits.Count; i++)
+        {
+            // this is the hsv identity of the unit we are currently dealing with
+            var identity = armyUnits[i].IdentityInfo;
+            Color.RGBToHSV(new Color(identity.r, identity.g, identity.b), out var h, out var s, out var v);
+
+            // this is fine?
+            // if both y and x are zero
+            // it just gives 0, which should have not effect in the subsequent calculation anyways
+            var angle = Mathf.Atan2(_armyComplex[i].y, _armyComplex[i].x);
+            
+            var newAngle = Mathf.MoveTowardsAngle(angle, meanAngle, _identityChangeRate * Time.fixedDeltaTime);
```

Comment:
Consider moving angles proportionally so that the *mean* color doesn't shift:

```suggestion
            var maxAngle = _armyComplex
                .Select(unitComplex => Mathf.Atan2(unitComplex.y, unitComplex.x))
                .Max();
            var maxAngleShift = Mathf.Abs(Mathf.MoveTowardsAngle(angle, meanAngle, _identityChangeRate * Time.fixedDeltaTime) - maxAngle);  // Cache this value
            var angleDelta = Mathf.Abs(Mathf.Atan2(_meanZ.y, _meanZ.x) - angle);
            var newAngle = Mathf.MoveTowardsAngle(angle, meanAngle, angleDelta / maxAngleShift);
```